src/fluree/db/flake/transact.cljc

```diff
@@ -61,12 +61,33 @@
 
 (defn into-flakeset
   [tracker error-ch flake-ch]
-  (let [flakeset (flake/sorted-set-by flake/cmp-flakes-spot)
-        error-xf (halt-when util/exception?)
-        flake-xf (if-let [track-fuel (track/track-fuel! tracker error-ch)]
-                   (comp error-xf track-fuel)
-                   error-xf)]
-    (async/transduce flake-xf (completing conj) flakeset flake-ch)))
+  (let [flakeset    (flake/sorted-set-by flake/cmp-flakes-spot)
+        error-xf    (halt-when util/exception?)
+        flake-xf    (if-let [track-fuel (track/track-fuel! tracker error-ch)]
+                      (comp error-xf track-fuel)
+                      error-xf)
+        retractions (volatile! [])]
+    (async/transduce
+     flake-xf
+     (completing
+      (fn [acc f]
```

Comment:
This is minor so you can take or leave it, but I think `acc` is too generic. The completion function assumes that `acc` is going to be a set of flakes, so `flakeset` makes that more explicit I think.